source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -337,31 +337,57 @@ public async ValueTask<string> AllocateIdAsync(string tableName, string idPrefix
             var key = await keyAllocator.NextIdAsync(tableName, cancellationToken).ConfigureAwait(false);
             return $"{idPrefix}-{key}";
         }
+        
+        public void CommitIfOwned()
+        {
+            if (!OwnsSqlTransaction) return;
+            
+            if (!configuration.AllowSynchronousOperations)
+                throw new SynchronousOperationsDisabledException();
+
+            CommitTransactionAndRunHooks();
+        }
 
         public void Commit()
         {
-            if (Transaction is null)
```

Comment:
There's comment in the matching [Server PR](https://github.com/OctopusDeploy/OctopusDeploy/pull/23120/files#diff-0bb7b050da17b51ac605821c161b7e5457452f6be49c1fb2afb72b5deea29467R239) that says Commit will no longer throw if the transaction is externally owned.

Did you mean to delete that `if` (below), rather than this `null` check?

Also, no-longer throwing an exception where one was thrown before is still a breaking change, although possibly one we can live with, and not a breaking change for server's usage.

Reply:
The comment in server was out of date and should have been removed.

This if that checks the transaction is not null has moved inside of `CommitTransactionAndRunHooks` to DRY things up.

> Also, no-longer throwing an exception where one was thrown before is still a breaking change, although possibly one we can live with, and not a breaking change for server's usage.

I dont believe we are no longer throwing exceptions here.... to make this a non breaking change I introduced `CommitIfOwnedAsync()` which is what UnitOfWork will call now.

Reply:
Ok, that makes sense then. Original behaviour of `Commit()` and `CommitAsync()` are now unchanged.
Looks good ✅